app/services/studio_cko_conversation_service.py

```
@@ -1655,6 +1773,164 @@ def _extract_cited_sources(
+
+            # Attach asset_id directly if memory was created
+            if memory_id:
+                from app.core.database import get_supabase
+                supabase = get_supabase()
+                await asyncio.to_thread(
+                    lambda: supabase.table("session_memories").update(
+                        {"asset_id": asset_id}
+                    ).eq("id", memory_id).execute()
+                )
+
+            return memory_id
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)